tests/src/core/testqgsexpression.cpp

```diff
@@ -975,6 +975,7 @@ class TestQgsExpression: public QObject
       // cast functions
       QTest::newRow( "double to int" ) << "toint(3.2)" << false << QVariant( 3 );
       QTest::newRow( "text to int" ) << "toint('53')" << false << QVariant( 53 );
+      QTest::newRow( "string double to int" ) << "toint('3.2')" << false << QVariant( 3 );
```

Comment:
Can you add test cases with non-dot locales such as italian or german?

Reply:
@elpaso isn't that breaking a fundamental design component of expressions? If I recall correctly the expression parser mandates dot-as-decimals. @wonder-sk do you have input here?

Reply:
@nyalldawson agreed: I asked for a test to make sure locales doesn't interfere, I would expect the same result with a non-dot locale.

Reply:
@nyalldawson now I see what you mean. The conversion is now locale-tolerant. Yes, I think expressions should use `C` locale. but I think that a locale-aware string to number conversion could be useful, I've seen many cases where a table contains locale-formatted numbers in a string field.

Reply:
+1 to stick to C locale in automatic conversion - otherwise it will get really confusing how it works in different parts of the code. Locale-aware conversion may be useful, but IMHO that should be done in a dedicated function (which would allow choosing locale as well - using the current system locale is a recipe for brittle code that works only on some computers)

Reply:
Should I change anything in this case?